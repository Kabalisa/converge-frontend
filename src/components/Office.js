import React from 'react';
import PropTypes from 'prop-types';
import DeleteOffice from './DeleteOffice'; //eslint-disable-line
import EditOffice from './EditOffice'; //eslint-disable-line

const Office = ({ office: { id, location, name }, refetch, currentPage }) => {
  let gmt = '';
  if (location.timeZone.includes('WEST')) {
    gmt = 'GMT +1';
  } else {
    gmt = 'GMT -1';
  }
  return (
    <tr>
      <td>{name}</td>
      <td>{location.name}</td>
      <td>{gmt}</td>
      <td>
        <EditOffice
          id="edit-modal"
          officeName={name}
          officeId={id}
          officeLocation={location.name}
          refetch={refetch}
          currentPage={currentPage}
        />
        &nbsp;
        <DeleteOffice
          officeName={name}
          id="delete-modal"
          officeId={id}
          refetch={refetch}
          currentPage={currentPage}
        />
      </td>
    </tr>
  );
};

Office.propTypes = {
  office: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    location: PropTypes.shape({
      name: PropTypes.string,
      timeZone: PropTypes.string,
    }),
  }).isRequired,
  refetch: PropTypes.func.isRequired,
  currentPage: PropTypes.number,
};

Office.defaultProps = {
  currentPage: 1,
};

export default Office;
