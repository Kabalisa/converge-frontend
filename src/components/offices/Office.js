import React from 'react';
import PropTypes from 'prop-types';
import DeleteOffice from './DeleteOffice'; //eslint-disable-line
import EditOffice from './EditOffice'; //eslint-disable-line

/**
 * shows offices data in a table
 *
 * @param {Object} office
 *
 * @returns {JSX}
 */
const Office = ({ office: { id, location, name }, refetch, currentPage }) => {
  let gmt = '';
  if (location.timeZone.includes('WEST')) {
    gmt = 'GMT +1';
  } else {
    gmt = 'GMT +3';
  }
  return (
    <div className="table__row">
      <span>{name}</span>
      <span>{location.name}</span>
      <span>{gmt}</span>
      <span>
        <EditOffice
          id="edit-modal"
          officeName={name}
          officeId={id}
          officeLocation={location.name}
          refetch={refetch}
          currentPage={currentPage}
        />
        &nbsp;
        <DeleteOffice
          officeName={name}
          id="delete-modal"
          officeId={id}
          refetch={refetch}
          currentPage={currentPage}
        />
      </span>
    </div>
  );
};

Office.propTypes = {
  office: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    location: PropTypes.shape({
      name: PropTypes.string,
      timeZone: PropTypes.string,
    }),
  }).isRequired,
  refetch: PropTypes.func.isRequired,
  currentPage: PropTypes.number,
};

Office.defaultProps = {
  currentPage: 1,
};

export default Office;
