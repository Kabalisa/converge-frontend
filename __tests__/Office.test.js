import React from 'react';
import { shallow } from 'enzyme';

import Office from '../src/components/Office';

describe('Tests for SettingOffices', () => {
  const office = {
    name: 'EPIC Tower',
    location: 'Nigeria',
    timezone: 'GMT +1',
  };

  const shalloWrapper = shallow(<Office office={office} />);

  it('renders correctly from memory', () => {
    expect(shalloWrapper).toMatchSnapshot();
  });
});
