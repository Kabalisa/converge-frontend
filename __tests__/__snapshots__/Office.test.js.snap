// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for SettingOffices renders correctly from memory 1`] = `
<tr>
  <td>
    EPIC Tower
  </td>
  <td>
    Nigeria
  </td>
  <td>
    GMT +1
  </td>
  <td>
    Edit â€ƒ Delete
  </td>
</tr>
`;
